### PR TITLE
[NFC] add comments & extract function in contribution search

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
This is the NFC part of #9631 - I think the main part can wait until after the cut off but the comment additions & database extraction allow simplification of that PR for review.

Note the only non-comment & non-test change is to extract the setting on soft credit options into it's own function

### CODE REVIEW

**Review by seamuslee001 — APPROVED** [review]
Looks good and my search still reports sensible things

**Comment by eileenmcnaughton** [comment]
Merging per review from Seamus.

File: CRM/Contact/BAO/Query.php [hunk 1]
```diff
@@ -588,14 +592,18 @@ public function buildParamsLookup() {
...
       }
     }
   }
 
   /**
    * Some composite fields do not appear in the fields array hack to make them part of the query.
+   *
+   * @param $apiEntity
+   *   The api entity being called.
+   *   This sort-of duplicates $mode in a confusing way. Probably not by design.
```
Comment (totten) [inline CRM/Contact/BAO/Query.php]: This looks nicer.

A suggestion on docblock style going-forward (not meant as a complaint about this): `@param $apiEntity` looks like a typical case where I'd try to include an example as part of the doc, as in:

```
 * @param string $apiEntity
 *   The api entity being called. Ex: 'Contact' or 'ContributionPage'.
```

As readers who are tuned-in to the nuances of Civi's data modelling (SQL/DAO/BAO/API), you and I probably interpret `apiEntity` as "an entity name in the camel-case notation of APIv3" (`ContributionPage`), but less experienced contributors might interpret "entity" more ambiguously. (`ContributionPage`, `contribution_page`, `api/v3/ContributionPage.php`, `civicrm_api3_contribution_page_*`, `CRM_Contribute_DAO_ContributionPage`, and `civicrm_contribution_page` are all ways to identify the same entity... and all those ways would actually show up if you were tracing an API call.)

An example value is a really simple/clear.

Comment (eileenmcnaughton) [inline CRM/Contact/BAO/Query.php]: Agree - although when I add stuff to pre-existing comments I don't feel the need to be that thorough

**Pull request merged**